src/app/login-modal/login-modal.component.ts

```diff
@@ -60,27 +70,58 @@ export class LoginModalComponent implements OnInit {
       return;
     }
 
-    this.loggingIn = true;
+    if (this.loggingIn) {
+      const credentials: Credentials = {
+        captcha_code: captchaResponse,
+        email_address: this.loginForm.value.emailAddress,
+        raw_password: this.loginForm.value.password,
+      };
+  
+      this.identityService.login(credentials).subscribe((response: { id: string, jwt: string }) => {
+        this.identityService.saveJWT(response.id, response.jwt);
+        this.dialogRef.close(response);
+        this.loggingIn = false;
+      }, (error) => {
+        this.captcha.reset();
+        this.loginError = (error.error.description !== undefined ? error.error.description : error.message) || 'Unknown error';
+        this.loggingIn = false;
+      });
+    }
 
-    const credentials: Credentials = {
-      captcha_code: captchaResponse,
-      email_address: this.form.value.emailAddress,
-      raw_password: this.form.value.password,
-    };
+    else if (this.userAskedForResetLink) {
+      this.identityService.getResetPasswordToken(this.resetPasswordForm.value.emailAddress).subscribe((response) => {
+        // console.log(response);
+        // console.log(response.id);
+        // console.log(response.jwt);
+        // this.identityService.saveJWT(response.id, response.jwt);
+        // this.dialogRef.close(response);
+        // this.userAskedForResetLink = false;
```

Comment:
is this commented out code here for a reason? Can we either add a comment to explain it or remove it?

Reply:
I've removed the comments.